src/storageManager.ts

```diff
@@ -69,6 +69,25 @@ export function newStorageManager({moduleName, moduleType}: {
         [ACTIVITY_PARAM_STORAGE_TYPE]: storageType
       }))
     };
+
+    // Additional check to prevent user ID modules from storing data in a storage type
+    // not allowed by the publisher's configuration
+    if (result.valid && moduleType === MODULE_TYPE_UID) {
+      const {userIds = [], enforceStorageType} = config.getConfig('userSync') || {};
+      const submoduleConfig = userIds.find(({name}) => moduleName === name);
+      if (!submoduleConfig || !submoduleConfig.storage) {
+        return cb(result);
+      }
+
+      if (submoduleConfig.storage.type !== storageType) {
+        const logFn = enforceStorageType ? logError : logWarn;
+        if (enforceStorageType) {
+          result.valid = false;
+        }
+        logFn(`${moduleName} attempts to store data in ${storageType} while configuration allows ${submoduleConfig.storage.type}.`);
```

Comment:
This should be an activity control rule added by the userId module.

Reply:
Does that make it simpler? What's the argument for activity control vs just enforcing configuration? It doesn't feel like the other activity controls, this one would be quite unique. What's the exit strategy when we have complete module compliance.